fix: if serve: false and root is not defined, only allow sending files with absolute path

Uzlopak opened the pull request:
If serve is set to false, it only allows to send files specified with an absolute path. Or else sendFile makes no sense anymore, despite that we want to allow the use of sendFile.



<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/fastify/fastify/blob/main/CONTRIBUTING.md

By making a contribution to this project, I certify that:

* (a) The contribution was created in whole or in part by me and I
  have the right to submit it under the open source license
  indicated in the file; or

* (b) The contribution is based upon previous work that, to the best
  of my knowledge, is covered under an appropriate open source
  license and I have the right under that license to submit that
  work with modifications, whether created in whole or in part
  by me, under the same open source license (unless I am
  permitted to submit under a different license), as indicated
  in the file; or

* (c) The contribution was provided directly to me by some other
  person who certified (a), (b) or (c) and I have not modified
  it.

* (d) I understand and agree that this project and the contribution
  are public and that a record of the contribution (including all
  personal information I submit with it, including my sign-off) is
  maintained indefinitely and may be redistributed consistent with
  this project or the open source license(s) involved.
-->

#### Checklist

- [ ] run `npm run test && npm run benchmark --if-present`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/main/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/main/CODE_OF_CONDUCT.md)

Uzlopak committed:
fix: if serve: false and root is not defined, only allow sending files with absolute path

Uzlopak committed:
fix

ilteoood reviewed (commented):
LGTM

is2ei commented:
I think @climba03003 would be the right person, as he originally mentioned the security concern.

climba03003 reviewed (approved):
LGTM.

Uzlopak commented on index.js:
```diff
@@ -19,8 +19,10 @@ send.mime.default_type = 'application/octet-stream'
+  if (opts.serve !== false) {
+    opts.root = normalizeRoot(opts.root)
+    checkRootPathForErrors(fastify, opts.root)
+  }
```
I think this is wrong. If opts.serve is false i can still have a defined root.

Uzlopak committed:
fix

If serve is set explicitly to false then root is required
If root is set, it must be validated

is2ei reviewed (approved):
LGTM!
